src/igraph_error.c

```diff
@@ -289,3 +324,42 @@ igraph_set_warning_handler (igraph_warning_handler_t * new_handler) {
     igraph_i_warning_handler = new_handler;
     return previous_handler;
 }
+
+
+/***** Handling fatal errors *****/
+
+static IGRAPH_THREAD_LOCAL igraph_fatal_handler_t *igraph_i_fatal_handler = NULL;
+
+igraph_fatal_handler_t *igraph_set_fatal_handler(igraph_fatal_handler_t * new_handler) {
+    igraph_fatal_handler_t * previous_handler = igraph_i_fatal_handler;
+    igraph_i_fatal_handler = new_handler;
+    return previous_handler;
+}
+
+#ifndef USING_R
```

Comment:
Perhaps I'm missing something, but couldn't we get rid of the `USING_R` thing here? After all, the `igraph_abort` will never pose any problem to R since it will not call `abort`.

Reply:
No, there's an `fprintf` and that's also not allowed. 

Actually I removed all these, then realized that we had the `fprintf`, put them back, and added a comment above `igraph_abort` at the top of the file.

Reply:
To be accurate, it's not the `fprintf` but the `stderr`.

Reply:
Wouldn't it then be easier to put that `fprintf` statement in `USING_R` defs? For `R` it issues an `error` anyway.

Reply:
The easiest thing was not to change it :-) R is not using the entire function anyway. If you think that something needs to be improved, feel free to go ahead.